usnistgov/oar-dmp-angular-ui
# Migrate unit test to Jest

This checkin migrates unit test from Jasmine-Karma to Jest.

To test:

build the application normally.

Test all (from root folder)
npm run test

Test single file (from root folder)
npm run test basic-info.component.spec

Debug with Jest

1.	Place “debugger” to wherever you want to set the breakpoint
2.	Run following command in a console window (the command is from Jest trouble shooting website https://jestjs.io/docs/troubleshooting) in the root folder of a project or library to debug the project or library.

cd oar-dmp
node --inspect-brk ../node_modules/.bin/jest --runInBand --config ./jest.config.js

To debug a single file:

node --inspect-brk ../node_modules/.bin/jest –runInBand --config ./jest.config.js <filename>
 For example:
node --inspect-brk ../node_modules/.bin/jest --runInBand --config ./jest.config.js basic-info.component.spec

3.	Open Chrome inspect: chrome://inspect and click on “inspect” at the bottom. The inspect window will pop up then you can start debugging.

See more details in the attached file.
[Migrating unit test to Jest.docx](https://github.com/usnistgov/oar-dmp-angular-ui/files/11234182/Migrating.unit.test.to.Jest.docx)
